packages/react-native-sdk/docusaurus/docs/reactnative/03-core/08-native-permissions.mdx

```diff
@@ -88,84 +91,111 @@ const iOSProcessResultsAndSetToConfig = (
-        nextAppState === 'active'
-      ) {
-        await checkAndUpdatePermissions();
-      }
-    };
-    const subscription = AppState.addEventListener(
-      'change',
-      handleAppStateChange,
-    );
+In this step, we create a function called `checkAndUpdatePermissions`. This function will be responsible for checking permissions and also updating the status to the SDK using the functions created in step-1. This function will be used later to check permissions when the app moves to foreground from the background.
 
-    return () => {
-      AppState.removeEventListener('change', handleAppStateChange);
-    };
-  }, []);
-};
+```ts title=src/utils/checkAndUpdatePermissions.ts
+import { Platform } from 'react-native';
+import { PERMISSIONS, checkMultiple } from 'react-native-permissions';
+import { androidProcessPermissions, iosProcessPermissions } from 'src/utils/updatePermissions';
 
-const checkAndUpdatePermissions = async () => {
+export const checkAndUpdatePermissions = async () => {
   if (Platform.OS === 'ios') {
     // Check, update and sync permissions on iOS
     const results = await checkMultiple([
       PERMISSIONS.IOS.CAMERA,
       PERMISSIONS.IOS.MICROPHONE,
     ]);
     // Sync the permissions with the Stream Video SDK
-    iOSProcessResultsAndSetToConfig(results);
+    iosProcessPermissions(results);
   } else if (Platform.OS === 'android') {
     // Check, update and sync permissions on Android
     const results = await checkMultiple([
       PERMISSIONS.ANDROID.CAMERA,
       PERMISSIONS.ANDROID.RECORD_AUDIO,
     ]);
     // Sync the permissions with the Stream Video SDK
-    androidProcessResultsAndSetToConfig(results);
+    androidProcessPermissions(results);
   }
 };
+```
+
+## Step 4 - Create the hook to manage permissions
+
+In the step, we create the `useSyncPermissions` hook. As mentioned earlier, this hook will be used to request the app's permissions and inform them to the SDK. 
+
+Additional to updating permissions after requesting them, it is also necessary to check the status of permissions when app comes to foreground. This will ensure that the permissions are updated when the user comes back to the app from the background after changing the permissions in the native settings screen. In order to do this, we will also add a listener to the app state change event in this hook.
+
+```tsx
+import { useEffect, useRef } from 'react';
+import { AppState } from 'react-native';
+import { requestAndUpdatePermissions } from 'src/utils/requestAndUpdatePermissions';
+import { checkAndUpdatePermissions } from 'src/utils/checkAndUpdatePermissions';
 
-const requestAndUpdatePermissions = async () => {...};
-const androidProcessResultsAndSetToConfig = (results) => {...};
-const iOSProcessResultsAndSetToConfig = (results) => {...};
+export const useSyncPermissions = () => {
+  // request permissions on mount
+  useEffect(() => {
+    requestAndUpdatePermissions();
+  }, []);
```

Comment:
```suggestion
   useEffect(() => {
    const requestAndUpdate = async () => {
      await requestAndUpdatePermissions();
    };
    requestAndUpdate();
  }, []);
```

Since `requestAndUpdatePermissions` is asynchronous. Feel free to use an appropriate name here.

Reply:
both snippets do the same, nothing is different